cluster_pipeline.go

```diff
@@ -0,0 +1,130 @@
+package redis
+
+// ClusterPipeline is not thread-safe.
+type ClusterPipeline struct {
+	commandable
+
+	cmds   []Cmder
+	client *ClusterClient
+	closed bool
+}
+
+// ClusterPipeline creates a new pipeline which is able to execute commands
+// against multiple shards. Be aware that multi-pipelines are
+// not able to follow redirects and you may receive errors if a node
+// fails or if keys are moved before/during pipeline execution.
+func (c *ClusterClient) ClusterPipeline() *ClusterPipeline {
```

Comment:
Can we rename this to just `ClusterClient.Pipeline`? There is no need to repeat cluster twice...